addons/google_calendar/models/google_sync.py

```diff
@@ -177,13 +177,17 @@ def _sync_google2odoo(self, google_events: GoogleEvent, default_reminders=()):
             # This could be dangerous if google server time and odoo server time are different
             updated = parse(gevent.updated)
             odoo_record = self.browse(gevent.odoo_id(self.env))
-            # Use the record's write_date to apply Google updates only if they are newer than Odoo's write_date.
-            odoo_record_write_date = write_dates.get(odoo_record.id, odoo_record.write_date)
-            # Migration from 13.4 does not fill write_date. Therefore, we force the update from Google.
-            if not odoo_record_write_date or updated >= pytz.utc.localize(odoo_record_write_date):
-                vals = dict(self._odoo_values(gevent, default_reminders), need_sync=False)
-                odoo_record.with_context(dont_notify=True)._write_from_google(gevent, vals)
-                synced_records |= odoo_record
+            try:
+                # Use the record's write_date to apply Google updates only if they are newer than Odoo's write_date.
+                odoo_record_write_date = write_dates.get(odoo_record.id, odoo_record.write_date)
+                # Migration from 13.4 does not fill write_date. Therefore, we force the update from Google.
+                if not odoo_record_write_date or updated >= pytz.utc.localize(odoo_record_write_date):
+                    vals = dict(self._odoo_values(gevent, default_reminders), need_sync=False)
+                    odoo_record.with_context(dont_notify=True)._write_from_google(gevent, vals)
+                    synced_records |= odoo_record
+            except MissingError:
+                # The record has been deleted in the meantime. Nothing to sync.
+                pass
```

Comment:
Encapsulating everything in try/except is a bad practice and it has a cost. What I would do is check that the odoo_record exists before using it.
1) `odoo_record = self.browse(gevent.odoo_id(self.env)).exists()`
2) avoid doing things if you know they will fail.
```py
 if not odoo_record:
    continue
```